### DISCUSSION TITLE
ERROR fatal: fetch-pack: invalid index-pack output

### DISCUSSION THREAD

**Discussion by FreQRiDeR** [Homebrew/discussions#3785] [Getting started]
Mac-DaDDY ~ % brew update
fatal: fetch-pack: invalid index-pack output
fatal: fetch-pack: invalid index-pack output
fatal: fetch-pack: invalid index-pack output
Error: Fetching /usr/local/Homebrew failed!
Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask failed!
Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core failed!

How to fix? Many bottles, casks fail to launch now with ERROR...
terramoda@Mac-DaDDY ~ % xboard     
2022-10-13 19:47:58.848 xboard[18677:73564] *** WARNING: Method userSpaceScaleFactor in class NSView is deprecated on 10.7 and later. It should not be used in new applications. Use convertRectToBacking: instead. 
zsh: illegal hardware instruction  xboard

Was working fine until I tried updating Java, I think? Not sure?

**Comment by p-linnane** [comment]
What is the output of `brew config` & `brew doctor`?

**Reply by FreQRiDeR** [reply]
Last login: Fri Oct 14 20:18:03 on console
terramoda@Mac-DaDDY ~ % brew config
HOMEBREW_VERSION: 3.6.4-15-gf35f9c0
ORIGIN: https://github.com/Homebrew/brew
HEAD: f35f9c03bef95fb1cf164b340392f7983933311f
Last commit: 12 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 125e81a717d7de6ebb2f5391968a4e457d110569
Core tap last commit: 11 days ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit penryn
Clang: 13.1.6 build 1316
Git: 2.37.3 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.4-x86_64
CLT: 13.4.0.0.1.1651278267
Xcode: N/A
terramoda@Mac-DaDDY ~ %

**Reply by FreQRiDeR** [reply]
terramoda@Mac-DaDDY ~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
terramoda@Mac-DaDDY ~ %

**Reply by p-linnane** [reply]
What happens when you run `brew update-reset && brew update`?

**Reply by p-linnane** [reply]
>2022-10-13 19:47:58.848 xboard[18677:73564] *** WARNING: Method userSpaceScaleFactor in class NSView is deprecated on 10.7 and later. It should not be used in new applications. Use convertRectToBacking: instead.

Given the latest stable is 6 years old, I'm not sure that's going to be corrected by upstream.

**Reply by FreQRiDeR** [reply]
Xboard was working fine for many months unril a few days ago. I think Java 18 messed something up because I'm getting illegal instructions error on many previously working packages. Xboard, htop, permasigner, and others error out now. I tried installing Java with Temurin cask but removed it when it didn't solve another issue. Seems like I have some AVX instuctions somewhere which my cpus don't support. Bizarre. Will post brew update --reset log when I get a chance. Thanks!

**Comment by FreQRiDeR** [comment]
SOLVED! Pyenv was misconfigured. Changing Python Global to 3.10 fixed the issue.

**Reply by FreQRiDeR** [reply]
ReInstalling Adoptium Java broke the same packages again. Any version. Even after removing /Library/Java/Virtual Machines/ it persists. I am getting an "illegal hardware instructions" error now when running certain things like htop, for example. This is on Monterey. 

I will try installing latest java on Mojave as these brew packages are working fine and see if it breaks things there.

**Reply by carlocab** [reply]
Are you on a Hackintosh?

**Comment by FreQRiDeR** [comment]
No, but I'm on an unsupported macPro 3,1 running openCore and Monterey so kind of a hackintosh. Lol
